sdk/core/core-http/src/policies/proxyPolicy.ts

```diff
@@ -12,29 +12,70 @@ import { ProxySettings } from "../serviceClient";
 import { WebResourceLike } from "../webResource";
 import { Constants } from "../util/constants";
 import { URLBuilder } from "../url";
+import { getEnvironmentValue } from "../util/utils";
+
+let noProxyList: any = null;
 
 function loadEnvironmentProxyValue(): string | undefined {
   if (!process) {
     return undefined;
   }
 
-  if (process.env[Constants.HTTPS_PROXY]) {
-    return process.env[Constants.HTTPS_PROXY];
-  } else if (process.env[Constants.HTTPS_PROXY.toLowerCase()]) {
-    return process.env[Constants.HTTPS_PROXY.toLowerCase()];
-  } else if (process.env[Constants.HTTP_PROXY]) {
-    return process.env[Constants.HTTP_PROXY];
-  } else if (process.env[Constants.HTTP_PROXY.toLowerCase()]) {
-    return process.env[Constants.HTTP_PROXY.toLowerCase()];
+  let httpsProxy =
+    getEnvironmentValue(Constants.HTTPS_PROXY) || getEnvironmentValue(Constants.ALL_PROXY);
+  let httpProxy =
+    getEnvironmentValue(Constants.HTTP_PROXY) || getEnvironmentValue(Constants.ALL_PROXY);
+
+  return httpsProxy || httpProxy;
+}
+
+export function isBypassed(uri: string) {
+  loadNoProxy();
+  for (let s of noProxyList) {
+    if (s[0] == ".") {
+      if (uri.endsWith(s)) {
+        return true;
+      } else {
+        uri = getHost(uri);
+        if (uri == s.slice(1) && uri.length == s.length - 1) {
+          return true;
+        }
+      }
+    } else {
+      if (uri == s) {
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
+function getHost(uri: string) {
```

Comment:
1. There has to be a js/ts lib that does this for you.
2. If not, then it should be in generic lib.

@jeremymeng  might know of one.

Reply:
We have `URLBuilder.parse(url).getHost()` in core-http.